components/script/dom/dedicatedworkerglobalscope.rs

```diff
@@ -70,20 +71,61 @@ impl<'a> Drop for AutoWorkerReset<'a> {
 }
 
 enum MixedMessage {
-    FromWorker((TrustedWorkerAddress, WorkerScriptMsg)),
+    FromWorker(WorkerScriptMsg),
     FromScheduler((TrustedWorkerAddress, TimerEvent)),
     FromDevtools(DevtoolScriptControlMsg)
 }
 
+impl CommonScriptMsgVariant for WorkerScriptMsg {
+    fn common_script_msg(&self) -> Option<&CommonScriptMsg> {
+        match self {
+            WorkerScriptMsg::Common(script_msg) => Some(script_msg),
+            _ => None
+        }
+    }
+
+    fn into_common_script_msg(self) -> Option<CommonScriptMsg> {
+        match self {
+            WorkerScriptMsg::Common(script_msg) => Some(script_msg),
+            _ => None
+        }
+    }
+
+    fn set_common_script_msg(script_msg: CommonScriptMsg) -> Self {
+        WorkerScriptMsg::Common(script_msg)
+    }
+}
+
+struct DedicatedWorkerPort {
+    receiver: Receiver<(TrustedWorkerAddress, WorkerScriptMsg)>,
+    dedicated_scope: DomRoot<DedicatedWorkerGlobalScope>,
+}
+
+impl TaskPort<(TrustedWorkerAddress, WorkerScriptMsg), WorkerScriptMsg> for DedicatedWorkerPort {
+    fn recv_tasks(&self) -> Vec<WorkerScriptMsg> {
+        self.receiver.try_iter().map(|msg|{
+            let (linked_worker, script_msg) = msg;
+            let _ar = AutoWorkerReset::new(&*self.dedicated_scope, linked_worker);
```

Comment:
I think this is actually not going to work, since we should only do the `AutoWorkerReset` 'right before' handling a particular event...